src/transform/src/analysis/equivalences.rs

```diff
@@ -297,30 +297,39 @@ pub struct EquivalenceClasses {
 }
 
 impl EquivalenceClasses {
+    /// Comparator function for the complexity of scalar expressions. Simpler expressions are
+    /// smaller. Can be used when we need to decide which of several equivalent expressions to use.
+    pub fn mir_scalar_expr_complexity(
+        e1: &MirScalarExpr,
+        e2: &MirScalarExpr,
+    ) -> std::cmp::Ordering {
+        use std::cmp::Ordering::*;
+        use MirScalarExpr::*;
+        match (e1, e2) {
+            (Literal(_, _), Literal(_, _)) => e1.cmp(e2),
+            (Literal(_, _), _) => Less,
+            (_, Literal(_, _)) => Greater,
+            (Column(_), Column(_)) => e1.cmp(e2),
+            (Column(_), _) => Less,
+            (_, Column(_)) => Greater,
+            (x, y) => {
+                // General expressions should be ordered by their size,
+                // to ensure we only simplify expressions by substitution.
+                // If same size, then fall back to the expressions' Ord.
+                match (x.size(), y.size()) {
+                    (x_size, y_size) if x_size == y_size => x.cmp(y),
+                    (x_size, y_size) => x_size.cmp(&y_size),
+                }
```

Comment:
```suggestion
                match x.size().cmp(&y.size()) {
                    Equal => x.cmp(y),
                    other => other,
                }
```

Reply:
Ah yes, thanks!

Reply:
It's even easier! `(x.size(), x).cmp(&(y.size(), y))`.